hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/ProtobufLogWriter.java

```diff
@@ -46,6 +46,10 @@
 
   protected FSDataOutputStream output;
 
+  private volatile long syncedLength = 0;
```

Comment:
Use an AtomicLong?

Reply:
seems that using AtomicLong is unnecessary, because AtomicLong could not provide `update if greater than` semantics,  so I used `synchronized` keyword here when updating the syncedLength for simplicity

Reply:
Use AtomicUtils.updateMax. It is a util class in hbase-common.

Reply:
Used `AtomicUtils` to replace `synchronized`, thank you very much.

Reply:
nit: why do we have `AtomicUtils.updateMax`? It seems [`getAndAccumulate`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLong.html#getAndAccumulate-long-java.util.function.LongBinaryOperator-) is designed for this use case, i.e., `syncedLength.getAndAccumulate(fsdos.getPos(), Math::max)`